ratings/models.py

```diff
@@ -42,3 +46,29 @@ def __unicode__(self):
     class Meta:
         unique_together = (('user', 'content_type', 'object_id'),)
         ordering = ('-created',)
+
+
+@receiver(post_delete, sender=Rating)
+def post_delete_rating(sender, instance, **kwargs):
+    try:
+        instance.content_object.num_ratings = F('num_ratings') - 1
+        rating = Rating.objects.filter(
+                content_type_id=instance.content_type_id,
+                object_id=instance.object_id).aggregate(Avg('rating')).values()[0]
+        if rating == None:
+            rating = 0
+        instance.content_object.avg_rating = rating
+        instance.content_object.save()
+    except ObjectDoesNotExist:
+        pass
+
+
+@receiver(post_save, sender=Rating)
+def update_num_ratings_on_post_insert(**kwargs):
+    instance = kwargs['instance']
```

Comment:
There are some things that we should change here, but with #979 I don't think it's worth changing them twice. We should merge both patches before we do a release @ffont